server/src/main/java/org/apache/iotdb/db/auth/authorizer/BasicAuthorizer.java

```diff
@@ -404,4 +410,105 @@ public void replaceAllUsers(Map<String, User> users) throws AuthException {
+      BasicStructureSerDeUtil.write(user.getPrivilegeList().size(), buffer);
+      for (PathPrivilege pathPrivilege : user.getPrivilegeList()) {
+        BasicStructureSerDeUtil.writeIntSet(pathPrivilege.getPrivileges(), buffer);
+        BasicStructureSerDeUtil.write(pathPrivilege.getPath(), buffer);
+      }
+      BasicStructureSerDeUtil.writeStringList(user.getRoleList(), buffer);
+    }
+
+    // serialize roleinfo
+    BasicStructureSerDeUtil.write(roleMap.size(), buffer);
+    for (String roleName : roleMap.keySet()) {
+      role = roleMap.get(roleName);
+      BasicStructureSerDeUtil.write(role.getName(), buffer);
+      for (PathPrivilege pathPrivilege : role.getPrivilegeList()) {
+        BasicStructureSerDeUtil.writeIntSet(pathPrivilege.getPrivileges(), buffer);
+        BasicStructureSerDeUtil.write(pathPrivilege.getPath(), buffer);
+      }
+    }
+    return buffer;
+  }
+
+  public void deserialize(ByteBuffer buffer) {
+    Map<String, User> userMap = new HashMap<>();
+    Map<String, Role> roleMap = new HashMap<>();
+    String username;
+    String password;
+    List<PathPrivilege> privilegeList = null;
+    Set<Integer> privilegesId = null;
+    String path = null;
+    List<String> roleList;
+    String roleName;
+
+    // deserialize user
+    int userMapSize = BasicStructureSerDeUtil.readInt(buffer);
+    for (int i = 0; i < userMapSize; i++) {
+      username = BasicStructureSerDeUtil.readString(buffer);
+      password = BasicStructureSerDeUtil.readString(buffer);
+      int privilegesSize = BasicStructureSerDeUtil.readInt(buffer);
+      if (privilegesSize == 0) {
+        privilegeList = new ArrayList<>();
+      } else {
+        for (int m = 0; m < privilegesSize; m++) {
+          privilegesId = BasicStructureSerDeUtil.readIntSet(buffer);
+          path = BasicStructureSerDeUtil.readString(buffer);
+        }
+        privilegeList.add(new PathPrivilege(privilegesId, path));
+      }
+      roleList = BasicStructureSerDeUtil.readStringList(buffer);
+      userMap.put(username, new User(username, password, privilegeList, roleList));
+    }
+
+    // deserialize role
+    int roleMapSize = BasicStructureSerDeUtil.readInt(buffer);
+    for (int i = 0; i < roleMapSize; i++) {
+      roleName = BasicStructureSerDeUtil.readString(buffer);
+      int privilegesSize = BasicStructureSerDeUtil.readInt(buffer);
+      if (privilegesSize == 0) {
+        privilegeList = new ArrayList<>();
+      } else {
+        for (int m = 0; m < privilegesSize; m++) {
+          privilegesId = BasicStructureSerDeUtil.readIntSet(buffer);
+          path = BasicStructureSerDeUtil.readString(buffer);
+        }
+        privilegeList.add(new PathPrivilege(privilegesId, path));
+      }
+      roleMap.put(roleName, new Role(roleName, privilegeList));
+    }
+
+    ((BasicUserManager) userManager).setUserMap(userMap);
+    ((BasicRoleManager) roleManager).setRoleMap(roleMap);
+  }
```

Comment:
Please add serialize and deserialize tests for BasicAuthorizer.

Reply:
fix